Dynamic Dejure Calculation

DrDudelsack opened the pull request:
Rather then having the same 769 start historical dejure setup, dejure kingdoms and empires are now calculated and created dynamically based on culture and rank. Duchies are calculated to be dejure in a kingdom tier title based off of province ownership, the title with the most provinces in a duchy will have that entire duchy be considered dejure. 

Independent kingdoms will have a dejure empire liege generated based off of their culture, as defined in dejureConversion.txt, while empire tier titles will be considered their own dejure empires, with governorship vassals being dejure kingdoms. Empire tier titles now get an additional kingdom tier title in their directly held land, named after the capital province, used primarily to avoid creating a hole in the generated dejure map. 

As an example, if Gaul from an I:R save remains divided, there will be many dejure kingdoms that are all dejure of a unitable Gallic Empire title, while if Boi conquers all of Italia and becomes an empire, the duchies in Italia will be dejure to Boi. This dejure generation prevents many cases where countries like Rome would ahistorically userp titles that shouldn't exist, such as France or Italy

DrDudelsack committed:
Add files via upload

Files used for default dejure titles, only come into play if they didn't exist in the I:R save

DrDudelsack committed:
Support for dynamic dejure generation

DrDudelsack committed:
Support for dynamic dejure generation

Added the dejureTitleCreation() method, which outputs dejure relationships to titlesDejure.txt

DrDudelsack committed:
Dynamic dejure creation support

importDejureList(), which imports dejure culture mappings found in dejureConversion.txt

importDuchyNameList(), which imports a list of all dejure duchies, along with the counties they are assigned to

DrDudelsack committed:
Dynamic Dejure calculations

Adds methods used for dynamically calculating the dejure relationship of titles based off of the I:R save file

DrDudelsack committed:
Added mapping for Khotan

DrDudelsack committed:
Initial Localization for default dejure titles

DrDudelsack committed:
Dynamic culture to dejure liege mapping file

Idhrendur reviewed (changes requested):
Codacy found some issues you should address: https://app.codacy.com/gh/ParadoxGameConverters/ImperatorToCk2/pullRequest?prid=7700691

DrDudelsack committed:
Copy dejureConversion.txt

I realized that I previously forgot to set the new file, dejureConversion.txt, to be copied in Visual Studio. Now it's been set to copy

DrDudelsack committed:
Formatting

DrDudelsack committed:
Bugfix

DrDudelsack committed:
Bugfix
